portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java

```diff
@@ -14,11 +14,253 @@
+
+		for (Country country : countries) {
+			countryLocalService.deleteCountry(country);
+		}
+	}
+
+	@Override
+	@SystemEvent(type = SystemEventConstants.TYPE_DELETE)
+	public Country deleteCountry(Country country) {
+
+		// Country
+
+		countryPersistence.remove(country);
+
+		// Regions
+
+		long countryId = country.getCountryId();
+
+		regionPersistence.removeByCountryId(countryId);
+
+		// Addresses
+
+		addressLocalService.deleteCountryAddresses(countryId);
+
+		// Organizations
+
+		List<Organization> organizationList = organizationLocalService.search(
+			country.getCompanyId(),
+			OrganizationConstants.ANY_PARENT_ORGANIZATION_ID, null, null, null,
+			countryId, null, QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+		for (Organization organization : organizationList) {
+			organization.setCountryId(0);
+			organization.setRegionId(0);
+
+			organizationLocalService.updateOrganization(organization);
+		}
+
+		return country;
+	}
+
+	@Override
+	public Country deleteCountry(long countryId) throws PortalException {
+		Country country = countryPersistence.findByPrimaryKey(countryId);
+
+		return countryLocalService.deleteCountry(country);
+	}
+
+	@Override
+	public Country fetchCountryByCompanyIdAndA2(long companyId, String a2) {
+		return countryPersistence.fetchByC_A2(companyId, a2);
+	}
+
+	@Override
+	public Country fetchCountryByCompanyIdAndNumber(
+		long companyId, String number) {
+
+		return countryPersistence.fetchByC_Number(companyId, number);
+	}
+
+	@Override
+	public List<Country> getCountriesByCompanyId(long companyId) {
+		return countryPersistence.findByCompanyId(companyId);
+	}
+
+	@Override
+	public List<Country> getCountriesByCompanyId(
+		long companyId, int start, int end,
+		OrderByComparator<Country> orderByComparator) {
+
+		return countryPersistence.findByCompanyId(
+			companyId, start, end, orderByComparator);
+	}
+
+	@Override
+	public List<Country> getCountriesByCompanyIdAndActive(
+		long companyId, boolean active, int start, int end,
+		OrderByComparator<Country> orderByComparator) {
+
+		return countryPersistence.findByC_Active(
+			companyId, active, start, end, orderByComparator);
+	}
+
+	@Override
+	public int getCountriesCountByCompanyId(long companyId) {
+		return countryPersistence.countByCompanyId(companyId);
+	}
+
+	@Override
+	public int getCountriesCountByCompanyIdAndActive(
+		long companyId, boolean active) {
+
+		return countryPersistence.countByC_Active(companyId, active);
+	}
+
+	@Override
+	public Country getCountryByCompanyIdAndA2(long companyId, String a2)
+		throws PortalException {
+
+		return countryPersistence.findByC_A2(companyId, a2);
+	}
+
+	@Override
+	public Country setActive(long countryId, boolean active)
+		throws PortalException {
+
+		Country country = countryPersistence.findByPrimaryKey(countryId);
+
+		country.setActive(active);
+
+		return countryPersistence.update(country);
+	}
+
+	@Override
+	public Country updateCommerceCountryChannelFilter(
```

Comment:
Our service doesn't have a channel concept. This method should be named `updateGroupFilterEnabled`.